pygpcca/_gpcca.py

```diff
@@ -1102,13 +1119,36 @@ def schur_matrix(self) -> OArray:
     @property  # type: ignore[misc]
     @d.dedent
     def top_eigenvalues(self) -> OArray:
-        """%(eigenvalues_m)s"""  # noqa: D400
+        """
+        Top :attr:`n_macrostates` eigenvalues.
+
+        %(eigenvalues_m)s
+        """
+        return self._top_eigenvalues
+
+    @property
+    def eigenvalues(self) -> OArray:
+        """Vector containing the eigenvalues of `P` associated to the requested cluster numbers."""
         return self._eigenvalues
 
     @property  # type: ignore[misc]
     @d.dedent
-    def cluster_crispness(self) -> Optional[float]:
-        """%(crispness_ret)s"""  # noqa: D400
+    def n_macrostates_crispness(self) -> Optional[float]:
+        """
+        Crispness for clustering into :attr:`n_macrostates` clusters.
+
+        %(crispness_ret)s
+        """
+        return self._crispness_opt
+
+    @property  # type: ignore[misc]
+    @d.dedent
+    def crispness_values(self) -> OArray:
```

Comment:
Maybe a better naming convention is needed here, you have `top_eigenvalues` <-> `n_macrostates_crispness` and `eigenvalues` <-> `crispness_values`, but it's not very obvious from the naming that one relates to the _optimal_ cluster number whereas the other is an iterable over `a range of cluster numbers`.

Reply:
Yes, it is difficult here, since we also accept the interval to only contain one cluster number and thus I wouldn't really call this optimal then...
But yes, strictly speaking even then it would be the optimal number out of an "interval" with only one number in it.
I don't really had good ideas here, mb. you guys @Marius1311 @michalk8  have nicer ones in mind?

Reply:
Tried to settle this [here](https://github.com/msmdev/pyGPCCA/pull/13/commits/08728698caaee7472b2dbd891118ada3a03ca37b).
@Marius1311 what is your opinion on this?

Reply:
Sorry, I don't get it, what's the new naming convention you propose?

Reply:
I not only renamed some properties, but also reordered all the properties. Mb. take a look [here](https://github.com/msmdev/pyGPCCA/blob/improve_optimize/pygpcca/_gpcca.py)